src/test/test-recovery-key.c

```diff
@@ -0,0 +1,158 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <string.h>
+
+#include "random-util.h"
+#include "recovery-key.h"
+#include "memory-util.h"
+#include "tests.h"
+
+TEST(make_recovery_key) {
+        _cleanup_(erase_and_freep) char *recovery_key = NULL;
+        size_t length;
+        const int num_test = 10;
+        char *generated_keys[num_test];
+        int r;
+
+        /* Check for succesful recovery-key creation */
+        r = make_recovery_key(&recovery_key);
+
+        assert_se(r == 0);
+        assert_se(recovery_key != NULL);
+        /* Check that length of formatted key is 72 with 64 modhex characters*/
+        length = strlen(recovery_key);
+
+        assert_se(length == RECOVERY_KEY_MODHEX_FORMATTED_LENGTH - 1);
```

Comment:
move the assert on the next line of the things they check, rather than leaving empty lines, here and elsewhere